src/timestamp9.c

```diff
@@ -219,40 +219,52 @@ timestamp9_in(PG_FUNCTION_ARGS)
 		/* it doesn't work - try our own simple parsing then */
 		struct tm tm_ = {0};
 		long long ns;
-		char plusmin;
-		char gmt_offset_str[6] = ""; /* length of XX:XX plus 1 according to sscanf rules to accomodate \0 */
+		char gmt_offset_str[TZ_STRLEN_MAX] = ""; /* length of XX:XX plus 1 according to sscanf rules to accomodate \0 */
 		int gmt_offset = 0;
 		int num_read;
 		time_t tt;
 
-		num_read = sscanf(str, "%d-%d-%d %d:%d:%d.%lld %c%5s", &tm_.tm_year, &tm_.tm_mon, &tm_.tm_mday, &tm_.tm_hour, &tm_.tm_min, &tm_.tm_sec, &ns, &plusmin, gmt_offset_str);
-		if (num_read == 9 && fractional_valid)
+		num_read = sscanf(str, "%d-%d-%d %d:%d:%d.%lld %s", &tm_.tm_year, &tm_.tm_mon, &tm_.tm_mday, &tm_.tm_hour, &tm_.tm_min, &tm_.tm_sec, &ns, gmt_offset_str);
```

Comment:
is there buffer overflow in case %s is longer than TZ_STRLEN_MAX?

Reply:
Good catch!